src/sirocco/parsing/_yaml_data_models.py

```diff
@@ -489,7 +564,7 @@ def get_plugin_from_named_base_model(
 ) -> str:
     if isinstance(data, (ConfigRootTask, ConfigShellTask, ConfigIconTask)):
         return data.plugin
-    name_and_specs = _NamedBaseModel.merge_name_and_specs(data)
+    name_and_specs = ConfigBaseTask.extract_merge_name(data)
```

Comment:
Since this is the `Data` class maybe we use the `ConfigBaseData`
```suggestion
    name_and_specs = ConfigBaseData.extract_merge_name(data)
```
I don't think this is important

Reply:
The argument of `get_plugin_from_named_base_model` actually is a task from which you want to get the `plugin`. The misunderstanding comes from the argument name, `data`, that should rather be `task`